coderd/insights.go

```diff
@@ -470,6 +470,67 @@ func (api *API) insightsTemplates(rw http.ResponseWriter, r *http.Request) {
 	httpapi.Write(ctx, rw, http.StatusOK, resp)
 }
 
+// @Summary Get insights about total users
+// @ID get-insights-about-total-users
+// @Security CoderSessionToken
+// @Produce json
+// @Tags Insights
+// @Param start_time query string true "Start time" format(date-time)
+// @Param end_time query string true "End time" format(date-time)
+// @Success 200 {object} codersdk.TotalUsersInsightResponse
+// @Router /insights/total-users [get]
+func (api *API) insightsTotalUsers(rw http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+
+	p := httpapi.NewQueryParamParser().
+		RequiredNotEmpty("start_time").
+		RequiredNotEmpty("end_time")
+	vals := r.URL.Query()
+	var (
+		// The QueryParamParser does not preserve timezone, so we need
+		// to parse the time ourselves.
+		startTimeString = p.String(vals, "", "start_time")
+		endTimeString   = p.String(vals, "", "end_time")
+	)
+	p.ErrorExcessParams(vals)
+	if len(p.Errors) > 0 {
+		httpapi.Write(ctx, rw, http.StatusBadRequest, codersdk.Response{
+			Message:     "Query parameters have invalid values.",
+			Validations: p.Errors,
+		})
+		return
+	}
+
+	startTime, endTime, ok := parseInsightsStartAndEndTime(ctx, rw, time.Now(), startTimeString, endTimeString)
+	if !ok {
+		return
+	}
+
+	zone, _ := startTime.Zone()
```

Comment:
You might find this in your tests too but Zone it's not guaranteed to return a name. It's unfortunately safer to use the offset (integer) and turn it into a string like `2 => "2"` and `-2 => "-2".

That's a bit unfortunate however since there may be a minor edge case during DST shift and could potentially produce an unwanted extra/missing row depending on the data. I haven't tried to verify if/how much of a problem it would be though.

Reply:
Changed to use offset!